http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=607426

From: Dominik Stadler <dominik.stadler@gmx.at>
Subject: mtools: Add support for zip-files to uz/lz

I created a small patch to mtools that enables support for zip-files for
uz/lz, which I find very handy as I now have one utility to look at and
extract archive files.

Index: mtools-4.0.18/scripts/uz
===================================================================
--- mtools-4.0.18.orig/scripts/uz
+++ mtools-4.0.18/scripts/uz
@@ -43,10 +43,12 @@ uzcmd='gzip -cd'
 case $0 in
 *uz)
 	tarparam="-pxvf"
+	zipparam=""
 	action="Extracting from "
 	;;
 *lz)
 	tarparam="-tvf"
+	zipparam="-l"
 	action="Reading directory of "
 	;;
 *)
@@ -65,13 +67,14 @@ while [ $# -ge 1 ]; do
 	echo >&2
 	found=
 
-	for suffix in "" .gz .tgz .tar.gz .z .tar.z .taz .tpz .Z .tar.Z .tar.bz2 tar.lz ; do
+	for suffix in "" .gz .tgz .tar.gz .z .tar.z .taz .tpz .Z .tar.Z .tar.bz2 tar.lz .zip .jar .war .ear .aar; do
 		if [ -r "${1}$suffix" ]; then
 			found=$1$suffix
 			break
 		fi
 	done
 
+	unzip=0
 	case $found in
 		*.tar.bz2 | *.tb2)
 			uzcmd='bzip2 -cd'
@@ -79,12 +82,19 @@ while [ $# -ge 1 ]; do
 		*.tar.lz)
 			uzcmd='lzip -cd'
 			;;
+		*.zip | *.jar | *.war | *.ear | *.aar)
+			unzip=1
+			;;
 	esac
 	if [ -z "$found" ]; then
 		echo "$0: could not read \"$1\"." >&2
 	else
 		echo "$action \"$found\"." >&2
-		$uzcmd -- "$found" | tar "$tarparam" -
+		if [ $unzip = 1 ]; then
+			unzip $zipparam -- "$found"
+		else
+			$uzcmd -- "$found" | tar "$tarparam" -
+		fi
 	fi
 	shift
 done
